framework/meta/src/cmd/contract/meta_config.rs

```diff
@@ -49,33 +48,64 @@ impl MetaConfig {
         }
     }
 
-    /// Cargo.toml files for secondary contracts are generated from the main contract Cargo.toml,
+    /// Cargo.toml files for all wasm crates are generated from the main contract Cargo.toml,
     /// by changing the package name.
-    pub fn generate_cargo_toml_for_secondary_contracts(&mut self) {
-        let main_contract = self.sc_config.main_contract_mut();
-
-        let main_cargo_toml_contents =
-            CargoTomlContents::load_from_file(main_contract.cargo_toml_path());
-        main_contract.wasm_crate_name = main_cargo_toml_contents.package_name();
-
-        for secondary_contract in self.sc_config.secondary_contracts() {
-            secondary_contract_cargo_toml(secondary_contract, &main_cargo_toml_contents)
-                .save_to_file(secondary_contract.cargo_toml_path());
+    pub fn generate_cargo_toml_for_all_wasm_crates(&mut self) {
+        let main_cargo_toml_contents = CargoTomlContents::load_from_file("../Cargo.toml");
+        let mut cargo_toml_data = WasmCargoTomlData::default();
```

Comment:
This is dangerous! State can jump from one variant to another.
The safe way:
Create a WasmCargoTomlData for each variant, in the loop.
No need for setter, you create it readonly from the beginning with all the necessary data.
You can extract the initialization to a separate function to make the code cleaner.